benchmarks/src/main/kotlin/bfschannel/BfsChannelBenchmark.kt

```diff
@@ -0,0 +1,320 @@
+        }
+    }
+}
+
+fun computeStandardDeviation(list : List<Long>) : Double {
+    var sum = 0.0
+    var standardDeviation = 0.0
+    for (num in list) {
+        sum += num
+    }
+    val mean = sum / list.size
+    for (num in list) {
+        standardDeviation += (num - mean).pow(2.0)
+    }
+    return sqrt(standardDeviation / list.size)
+}
+
+private fun runParallelBfs(startNode: Node, nodes: Int, workers: Int) : Pair<Long, List<Long>> {
+    val start = System.nanoTime()
+    val parallelDistances = bfsParallel(startNode, nodes, workers)
+    val end = System.nanoTime()
+
+    return Pair(end - start, parallelDistances)
+}
+
+private fun runSequentialBfs(startNode: Node, nodes: Int) : Pair<Long, List<Long>> {
+    val start = System.nanoTime()
+    val sequentialDistances = bfsSequential(startNode, nodes)
+    val end = System.nanoTime()
+
+    return Pair(end - start, sequentialDistances)
+}
+
+fun bfsSequential(startNode: Node, nodes: Int): List<Long> {
+    val distances = Array(nodes) { Long.MAX_VALUE }
+    // The distance to the start node is `0`
+    distances[startNode.id] = 0
+
+    val queue = LinkedList<Node>()
+    queue.add(startNode)
+    while (queue.isNotEmpty()) {
+        val currentNode = queue.poll()
+        for (edge in currentNode.outgoingEdges) {
+            val newDistance = distances[currentNode.id] + 1
+            val oldDistance = distances[edge.to.id]
+            if (newDistance < oldDistance) {
+                distances[edge.to.id] = newDistance
+                queue.add(edge.to)
+            }
+        }
+    }
+    return distances.toList()
+}
+
+fun bfsParallel(start: Node, nodes: Int, workers: Int): List<Long> {
+    val distances = Array(nodes) { AtomicLong(Long.MAX_VALUE) }
+    // The distance to the start node is `0`
+    distances[start.id].set(0)
+
+    val queue : Channel<Node> = SelfClosingChannel(workers)
+    queue.offer(start)
+    // Run worker threads and wait until the total work is done
+    val jobs = ArrayList<Job>()
+    repeat(workers) {
+        jobs += GlobalScope.launch {
+            while (true) {
+                val currentNode = queue.receiveOrClosed().valueOrNull ?: break
+                for (edge in currentNode.outgoingEdges) {
+                    processEdge(distances, currentNode, edge.to, queue)
+                }
+            }
+        }
+    }
+
+    runBlocking {
+        for (job in jobs) {
+            job.join()
+        }
+    }
+    // Return the result
+    return distances.map { long -> long.toLong() }
+}
+
+private fun processEdge(distances: Array<AtomicLong>, currentNode: Node, nextNode: Node, queue: Channel<Node>) {
```

Comment:
`processEdge` -> `relaxEdge`

Create a part of the benchmark class in order not to pass all these parameters